OrbitGgp/Client.cpp

```diff
@@ -86,43 +86,35 @@ outcome::result<Client> Client::Create() {
     ERROR("Ggp command line process failed with error: %s (exit code: %d)",
           ggp_process.errorString().toStdString().c_str(),
           ggp_process.exitCode());
-    return Error::kCouldNotUseGgpCli;
+    return nullptr;
```

Comment:
I thought it was unnecessarily complicated. Create as is, returns either a QPointer to a Client object when successful, or fails aka returns a nullptr. The only benefit of using outcome::result would be to have a more unified interface and I don't know if that is worth the extra code. Or am I missing something?

Reply:
I don't think you're missing something. For me it was all about the more unified interface. And I see your point, but I don't see where extra code is involved. Maybe I am missing something? :sweat_smile:

Reply:
With extra code I just mean the wrapping of `QPointer<Client>` into `outcome::result`. It   feels a little bit like unnecessary boilerplate. Which makes it take tiny little bit longer and harder to read. But I guess there would be the added benefit of making it more clear that this can fail.

Reply:
Yes that's how I feel about that. I would also wrap a raw pointer in a std::optional in similar cases. But that's also not a strong feeling. So you can choose whatever you like better.